scripts/next_step_assignment_spectramax.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+from EPPs.common import StepEPP
+from pyclarity_lims.entities import Protocol
+
+
+class AssignmentNextStepSpectramax(StepEPP):
+    """
+    This script assigns the next step to samples in the Spectramax Picogreen step at initial sample QC. 
+    Samples with output udf 'Repeat Picogreen with 1:10 Dilution?' as true will be directed to a dilution step.
+    Samples that have the sample udf 'Prep Workflow' as KAPA DNA Sample Prep will skip Fragment Analyser
+    and be queued to QC Review. 
+    Samples that have gone through the 1:10 Dilution and already completed the Fragment Analyser step will be queued 
+    to the QC Review step.
+    All other samples with will be queued to Fragment Analyser. 
+    """
+    _use_load_config = False  # prevent the loading of the config file
+
+    def _run(self):
+        # obtain the actions of the step then creates a StepActions entity for the current step
+        actions = self.process.step.actions
+
+        # obtain the next actions in the step then creates a list of dict for next_actions for the step
+        next_actions = actions.next_actions
+
+        # prepare the step_object variable used for inputs that move to the next step
+        current_step = self.process.step.configuration  # configuration gives the ProtocolStep entity.
+        protocol = Protocol(self.process.lims, uri='/'.join(self.process.step.configuration.uri.split('/')[:-2]))
+
+        steps = protocol.steps  # a list of all the ProtocolSteps in protocol
+
+        for next_action in next_actions:
+
+            if self.process.outputs_per_input(next_action['artifact'].id,ResultFile=True)[0].udf.get('Repeat Picogreen with 1:10 Dilution?'):
+                # update the next action to the third step from this step i.e. Picogreen 1-10 Dilution
+                next_step_object = steps[steps.index(current_step) + 3]
```

Comment:
We're relying here on the step order - if this changes and we forget to update this, we could get incorrect next step assignments without knowing. Is there any way of checking the step name?

Reply:
Updated to use the config file to specify the step object rather than using numbers in a list.